### ISSUE TITLE
Problem with cmor.write when trying to produce 5-dimensional variable

### ISSUE THREAD

**Issue by ehogan** [PCMDI/cmor#224]
I'm trying to produce a 5-dimensional variable ([cfadLidarsr532](https://github.com/PCMDI/cmip6-cmor-tables/blob/c3a33350bf243a20f7dbe2635951ca70e0d7a905/Tables/CMIP6_Emon.json#L772); `longitude latitude alt40 scatratio time`) but [cmor.write](https://github.com/PCMDI/cmor/blob/a857a29fe0b45cf4354ab13f12719f9838969bec/Lib/pywrapper.py#L729) raises the exception `Error: your data shape ((1, 15, 40, 144, 192)) does not match the expected variable shape ([0, 40, 144, 192])`. After some investigation, I discovered this occurred only when I supply `time_vals` and `time_bnds` to `cmor.write`; if these values are included when defining the time axis, CMOR raises no exception. I wrote a test module to demonstrate the behaviour. If you copy the code below into `Test/test_python_write.py` then run `nosetests Test/test_python_write.py`, the first test will fail and the second test will pass.

The [code here](https://github.com/PCMDI/cmor/blob/a857a29fe0b45cf4354ab13f12719f9838969bec/Src/cmor_variables.c#L1327-L1749) determines the expected variable shape as `[0, 40, 144, 192]`. This is the first time I have tried to produce a 5 dimensional variable, so I'm wondering if the logic here isn't quite working in this case? In addition, in the case where `time_vals` and `time_bnds` are provided to `cmor.write`, the [code here](https://github.com/PCMDI/cmor/blob/a857a29fe0b45cf4354ab13f12719f9838969bec/Lib/pywrapper.py#L706) sets `ntimes_passed = 1`, which ultimately results in the exception being raised.

Apologies for not getting any further with this; @dnadeau4 I'm hoping your knowledge of the logic here will enable you to resolve this quickly :)

```
# If this example is not executed from the directory containing the
# CMOR code, please first complete the following steps:
#
#   1. In any directory, create 'Tables/', 'Test/' and 'CMIP6/'
#      directories.
#
#   2. Download CMIP6_CV.json, CMIP6_Emon.json, CMIP6_coordinate.json
#      and CMIP6_formula_terms.json from
#      https://github.com/PCMDI/cmor/blob/master/TestTables/ to the
#      'Tables/' directory.
#
#   3. Download common_user_input.json from
#      https://github.com/PCMDI/cmor/blob/master/Test/ to the 'Test/'
#      directory.
#
# pylint: disable = missing-docstring, invalid-name, no-member, no-self-use
"""
Tests for ``cmor.write``.
"""
import numpy as np
import os
import unittest

import cmor


class TestWrite(unittest.TestCase):
    """
    Tests for ``cmor.write``.
    """

    def setUp(self):
        self.logfile = 'write.log'
        cmor.setup(inpath='Tables', logfile=self.logfile)
        cmor.dataset_json('Test/common_user_input.json')
        cmor.load_table('CMIP6_Emon.json')
        self.axis_names = [
            'time', 'scatratio', 'alt40', 'latitude', 'longitude']

    def test_write_with_time_vals_and_bnds(self):
        # Do not include time coordinate values and bounds when
        # defining the time axis.
        for name in ['time']:
            time_axis_id = [cmor.axis(name, units=self._units[name])]
            time_vals = np.array(self._coord_vals[name])
            time_bnds = np.stack([self._cell_bounds[name][:-1],
                                  self._cell_bounds[name][1:]], axis=1)
        self.axis_names.remove('time')
        other_axis_ids = self._cmor_axis(self.axis_names)
        axis_ids = time_axis_id + other_axis_ids
        self._cmor_variable_and_write(axis_ids, time_vals, time_bnds)

    def test_write_without_time_vals_and_bnds(self):
        # Include time coordinate values and bounds when defining the
        # time axis.
        axis_ids = self._cmor_axis(self.axis_names)
        self._cmor_variable_and_write(axis_ids, None, None)

    def _cmor_axis(self, axis_names):
        axis_ids = [
            cmor.axis(
                name, units=self._units[name],
                coord_vals=np.array(self._coord_vals[name]),
                cell_bounds=np.stack([self._cell_bounds[name][:-1],
                                      self._cell_bounds[name][1:]], axis=1))
            for name in axis_names]
        return axis_ids

    @property
    def _units(self):
        return {
            'longitude': 'degrees',
            'latitude': 'degrees',
            'alt40': 'm',
            'scatratio': '1',
            'time': 'days since 2000-01-01',
            }

    @property
    def _coord_vals(self):
        return {
            'longitude': [float(value) for value in range(0, 192)],
            'latitude': [float(value) for value in range(0, 144)],
            'alt40': [240., 720., 1200., 1680., 2160., 2640., 3120., 3600.,
                      4080., 4560., 5040., 5520., 6000., 6480., 6960., 7440.,
                      7920., 8400., 8880., 9360., 9840., 10320., 10800.,
                      11280., 11760., 12240., 12720., 13200., 13680., 14160.,
                      14640., 15120., 15600., 16080., 16560., 17040., 17520.,
                      18000., 18480., 18960.],
            'scatratio': [0.005, 0.605, 2.1, 4., 6., 8.5, 12.5, 17.5, 22.5,
                          27.5, 35., 45., 55., 70., 50040.],
            'time': [-7455.],
            }

    @property
    def _cell_bounds(self):
        return {
            'longitude': [float(value) for value in range(0, 193)],
            'latitude': [float(value) for value in range(0, 145)],
            'alt40': [0., 480., 960., 1440., 1920., 2400., 2880., 3360., 3840.,
                      4320., 4800., 5280., 5760., 6240., 6720., 7200., 7680.,
                      8160., 8640., 9120., 9600., 10080., 10560., 11040.,
                      11520., 12000., 12480., 12960., 13440., 13920., 14400.,
                      14880., 15360., 15840., 16320., 16800., 17280., 17760.,
                      18240., 18720., 19200.],
            'scatratio': [0., 0.01, 1.2, 3., 5., 7., 10., 15., 20., 25., 30.,
                          40., 50., 60., 80., 100000.,],
            'time': [-7470., -7440.],
            }

    def _cmor_variable_and_write(self, axis_ids, time_vals, time_bnds):
        var_id = cmor.variable('cfadLidarsr532', axis_ids=axis_ids, units='1')
        data = np.random.random(16588800)
        reshaped_data = data.reshape((1, 15, 40, 144, 192))
        cmor.write(var_id, reshaped_data, time_vals=time_vals,
                   time_bnds=time_bnds)

    def tearDown(self):
        cmor.close()
        if os.path.isfile(self.logfile):
            os.remove(self.logfile)


if __name__ == '__main__':
    unittest.main()
```

**Comment by dnadeau4** [comment]
@ehogan Thanks for the test, the code is not mine, but I will run it step by step in the debugger and should figure out what goes wrong.

**Comment by dnadeau4** [comment]
Fix in 3.2.6
